pkg/gofr/datasource/file/fs.go

```diff
@@ -61,3 +61,26 @@ func (fileSystem) RemoveAll(path string) error {
 func (fileSystem) Rename(oldname, newname string) error {
 	return os.Rename(oldname, newname)
 }
+
+func (fileSystem) Stat(name string) (FileInfo, error) {
```

Comment:
How can it return a FileInfo ?

I would have understand os.FileInfo

If a local fileinfo defined as an alias of os.FileInfo, and was available in the package and here there was a type juggling

Does it even compile?

What am I missing?

Reply:
I have added an interface FileInfo that has all the methods of the os.FileInfo. This is because I need to define FileInfo for the ftp Filesystem also .... so that it works for both the filesystems.

Reply:
Oh indeed, you return an interface that os.FileInfo satisfies 🤦‍♂️

Great then, thanks for the explanation